lib/Mojolicious/Plugin/Web/Auth/OAuth2.pm

```diff
@@ -116,7 +117,10 @@ sub _ua {
 
 sub _response_to_hash {
     my ( $self, $res ) = @_;
-    return ( $res->headers->content_type =~ /^application\/json[;]?/ )
+    my $types = Mojolicious::Types->new;
+    $types->type(json => ['application/json', 'text/javascript']);
+    my $exts = $types->detect( $res->headers->content_type );
+    return ( $exts->[0] && $exts->[0] eq 'json' )
```

Comment:
How about this?

```perl
return ( scalar(@$exts) && $exts->[0] eq 'json' )
```

Reply:
Good idea.  I've updated and force-pushed to the branch.